amgut/db/scrub.sql

```diff
@@ -0,0 +1,343 @@
+    -- Magic number 111: birth month question
+    FOR rec IN
+        SELECT survey_id FROM ag.survey_answers WHERE survey_question_id = 111
+    LOOP
+        -- Adapted from http://dba.stackexchange.com/a/55401
+        UPDATE ag.survey_answers
+            SET response = ('[0:11]={January,February,March,April,May,June,July,August,September,October,November,December}'::text[])[trunc(random()*12)]
+            WHERE survey_id = rec.survey_id AND survey_question_id = 111;
+    END LOOP;
+
+    -- Table: survey_answers_other; Columns: response
+    source := retrieve_chars('response', 'ag.survey_answers_other');
+    -- This column is stored in a weird way. It looks like a python list, but
+    -- it always have a single element, even if it is an empty string. Thus,
+    -- removing those characters that can break this structure
+    source := replace(replace(replace(source, '""', ''), '[', ''), ']', '');
+    FOR rec IN
+        SELECT survey_id, survey_question_id FROM ag.survey_answers_other
+    LOOP
+        UPDATE ag.survey_answers_other
+            SET response = '["' || random_string(20, source, 'Free text - ') || '"]'
+            WHERE survey_id = rec.survey_id AND survey_question_id = rec.survey_question_id;
+    END LOOP;
+
+    -- Table: ag_participant_exceptions; columns: participant_name
+    source := retrieve_chars('participant_name', 'ag.ag_participant_exceptions');
+    FOR rec IN
+        SELECT ag_login_id FROM ag.ag_participant_exceptions
+    LOOP
+        UPDATE ag.ag_participant_exceptions
+            SET participant_name = random_string(10, source, 'Name - ')
+        WHERE ag_login_id = rec.ag_login_id;
+    END LOOP;
+
+    -- Table: ag_animal_survey; Columns: participant_name, comments
+    source := retrieve_chars('participant_name', 'ag.ag_animal_survey');
+    source2 := retrieve_chars('comments', 'ag.ag_animal_survey');
+    FOR rec IN
+        SELECT ag_login_id, participant_name FROM ag.ag_animal_survey
+    LOOP
+        UPDATE ag.ag_animal_survey
+            SET participant_name = random_string(10, source, 'Name - '),
+                comments = random_string(50, source2, 'Free text - ')
+            WHERE ag_login_id = rec.ag_login_id AND participant_name = rec.participant_name;
+    END LOOP;
+
+    -- Table: ag_login; columns: email, name, address, city, state, zip, latitude, longitude
+    source := retrieve_chars('address', 'ag.ag_login');
+    source2 := retrieve_chars('city', 'ag.ag_login');
+    source3 := retrieve_chars('state', 'ag.ag_login');
+    FOR rec IN
+        SELECT ag_login_id FROM ag.ag_login
+    LOOP
+        UPDATE ag.ag_login
+            SET email = random_email('email', 'ag.ag_login'),
+                name = random_string('name', 'ag.ag_login'),
+                address = random_string(4, numsource, '') || random_string(10, source, ' '),
+                city = random_string(10, source2, 'City - '),
+                state = random_string(10, source3, 'State - '),
```

Comment:
Want to note that as a point of normalization? Not sure how that would happen so wonder if it's from the original db